src/core/InterleavedBuffer.js

```diff
@@ -89,9 +92,57 @@ Object.assign( InterleavedBuffer.prototype, {
 
 		return this;
 
+	},
+
+	toJSON: function ( data ) {
+
+		if ( data.arrayBuffers === undefined ) {
+
+			data.arrayBuffers = {};
+
+		}
+
+		// generate UUID for array buffer if necessary
+
+		if ( this.array.buffer._uuid === undefined ) {
+
+			this.array.buffer._uuid = MathUtils.generateUUID();
+
+		}
+
+		if ( data.arrayBuffers[ this.array.buffer._uuid ] === undefined ) {
+
+			data.arrayBuffers[ this.array.buffer._uuid ] = arrayBufferToBase64( this.array.buffer );
```

Comment:
Hello

I would consider storing this as regular numeric array data here. Opens up possibility for optimizations with binary based JSON compatible formats.

(https://github.com/mrdoob/three.js/pull/19233#issuecomment-637012839)

Thanks  a lot

Reply:
My idea was to use the same approach like embedded buffer data in glTF files.

@donmccurdy What do you recommend here?

Reply:
@tentone I've added b0cda0bbf1077424006f9a85c29f7bc229569d27 to implement your suggested approach. I'm using `Uint16Array` to save the data in order to avoid any precision lost due to floating point. What do you think?

Reply:
Oh sorry havent noticed that commit.

I think it would be better to declare the `TypedArray` exact type, would allow the serialization to be faster by removing the need to cast everything to `Uint32`,would save some memory on runtime, and allow for 64 bit types in the future.

Possibly something like this on the serialization side of things:

```
if ( data.arrayBuffers[ this.array.buffer._uuid ] === undefined ) {

	data.arrayBuffers[ this.array.buffer._uuid ] = {
		array: Array.prototype.slice.call( this.array.buffer );
		type: this.array.buffer.constructor.name, // e.g "Int8Array"
	};

}
```

What do you think of this approach?

Reply:
The thing is that different interleaved buffers can share the same array buffer. But you want to serialize the array buffer just once. Hence, you could use any type of array to represent the data or convert it to Base64 string. It's just a matter of what approach produces the smallest size and is the fastest.

Reply:
Oh i see so we might have the same `ArrayBuffer `used for different `TypedArrays` I assuming that the `TypedArray `would always be of the same type for each of the shared `ArrayBuffer`.

If thats the case the current approach by moving everything to `UInt32 `as you have done seems to be the better approach.